/*
 * This file was automatically generated by SmartUnit
 */

package sut.com.wanxin.uaa.common.intercept;

import org.junit.Test;
import static org.junit.Assert.*;
import java.lang.reflect.Method;
import static org.smartunit.shaded.org.mockito.Mockito.*;
import org.smartunit.shaded.org.mockito.ArgumentMatchers;
import com.wanxin.common.domain.BusinessException;
import com.wanxin.common.domain.ErrorCode;
import com.wanxin.common.domain.RestResponse;
import com.wanxin.uaa.common.intercept.GlobalExceptionHandler;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import org.junit.runner.RunWith;
import org.smartunit.runtime.ExecutorServiceAnswer;
import org.smartunit.runtime.SmartRunner;
import org.smartunit.runtime.SmartRunnerParameters;
import org.smartunit.runtime.TransactionTemplateAnswer;
import org.smartunit.runtime.ViolatedAssumptionAnswer;
import org.springframework.lang.Nullable;

@RunWith(SmartRunner.class) @SmartRunnerParameters(mockJVMNonDeterminism = true, useVNET = true, resetStaticState = true) 
public class GlobalExceptionHandler_SSTest extends GlobalExceptionHandler_SSTest_scaffolding {
// allCoveredLines:[25, 31, 32, 33, 36, 39, 41, 43, 45, 49, 50]

  @Test(timeout = 9000)
  public void test_exceptionGet_0()  throws Throwable  {
      //caseID:423133b3e6bb90e113a99107b97f6c58
      //CoveredLines: [25, 31, 32, 33, 36]
      //Input_0_HttpServletRequest: {}
      //Input_1_HttpServletResponse: {}
      //Input_2_Exception: errorCode0
      //Assert: assertEquals("\u81EA\u5B9A\u4E49\u5F02\u5E38", method_result.getMsg());
      
      GlobalExceptionHandler globalExceptionHandler0 = new GlobalExceptionHandler();
      //mock httpServletRequest0
      HttpServletRequest httpServletRequest0 = mock(HttpServletRequest.class, withSettings().defaultAnswer(new ViolatedAssumptionAnswer()).stubOnly());
      //mock httpServletResponse0
      HttpServletResponse httpServletResponse0 = mock(HttpServletResponse.class, withSettings().defaultAnswer(new ViolatedAssumptionAnswer()).stubOnly());
      //mock errorCode0
      ErrorCode errorCode0 = mock(ErrorCode.class, withSettings().defaultAnswer(new ViolatedAssumptionAnswer()).stubOnly());
      doReturn(999999).when(errorCode0).getCode();
      doReturn("\u81EA\u5B9A\u4E49\u5F02\u5E38").when(errorCode0).getDesc();
      BusinessException businessException0 = new BusinessException(errorCode0);
      
      //Call method: exceptionGet
      RestResponse<Nullable> restResponse0 = globalExceptionHandler0.exceptionGet(httpServletRequest0, httpServletResponse0, businessException0);
      
      //Test Result Assert
      assertEquals("\u81EA\u5B9A\u4E49\u5F02\u5E38", restResponse0.getMsg());
  }

  @Test(timeout = 9000)
  public void test_exceptionGet_1()  throws Throwable  {
      //caseID:0af6cdde318ddc5725df148c70ec6071
      //CoveredLines: [25, 31, 39, 41, 43, 45, 49, 50]
      //Input_0_HttpServletRequest: {}
      //Input_1_HttpServletResponse: {}
      //Input_2_Exception: {}
      //Assert: assertEquals(999999, method_result.getCode());
      
      GlobalExceptionHandler globalExceptionHandler0 = new GlobalExceptionHandler();
      //mock httpServletRequest0
      HttpServletRequest httpServletRequest0 = mock(HttpServletRequest.class, withSettings().defaultAnswer(new ViolatedAssumptionAnswer()).stubOnly());
      //mock httpServletResponse0
      HttpServletResponse httpServletResponse0 = mock(HttpServletResponse.class, withSettings().defaultAnswer(new ViolatedAssumptionAnswer()).stubOnly());
      //mock exception0
      Exception exception0 = mock(Exception.class, withSettings().defaultAnswer(new ViolatedAssumptionAnswer()).stubOnly());
      
      //Call method: exceptionGet
      RestResponse<Nullable> restResponse0 = globalExceptionHandler0.exceptionGet(httpServletRequest0, httpServletResponse0, exception0);
      
      //Test Result Assert
      assertEquals(999999, restResponse0.getCode());
  }
}
